File: src/ts/magma/app/compile/Context.ts
import { Source } from "../../../magma/app/io/Source";
import { Iter } from "../../../magma/api/collect/Iter";
import { Platform } from "../../../magma/app/Platform";
import { Option } from "../../../magma/api/option/Option";
import { Location } from "../../../magma/app/Location";
export interface Context {
	iterSources(): Iter<Source>;
	hasPlatform(platform: Platform): boolean;
	findSource(name: string): Option<Source>;
	withLocation(location: Location): Context;
	addSource(source: Source): Context;
	findLocation(): Option<Location>;
	withPlatform(platform: Platform): Context;
}

File: src/java/magma/app/compile/Context.java
package magma.app.compile;

import magma.api.collect.Iter;
import magma.api.option.Option;
import magma.app.Location;
import magma.app.Platform;
import magma.app.io.Source;

public interface Context {
    Iter<Source> iterSources();

    boolean hasPlatform(Platform platform);

    Option<Source> findSource(String name);

    Context withLocation(Location location);

    Context addSource(Source source);

    Option<Location> findLocation();

    Context withPlatform(Platform platform);
}
